sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1086 @@
+     */
+    IEEE802154_CAP_SUB_GHZ,
+    /**
+     * @brief the device reports when the transmission is done
+     */
+    IEEE802154_CAP_IRQ_TX_DONE,
+    /**
+     * @brief the device reports the start of a frame (SFD) when received.
+     */
+    IEEE802154_CAP_IRQ_RX_START,
+    /**
+     * @brief the device reports the end of the CCA procedure
+     */
+    IEEE802154_CAP_IRQ_CCA_DONE,
+    /**
+     * @brief the device supports going to RX after TX done.
+     */
+    IEEE802154_CAP_TXRX,
+    /**
+     * @brief the device provides the number of retransmissions
+     */
+    IEEE802154_CAP_FRAME_RETRIES_INFO,
+} ieee802154_rf_caps_t;
+
+/**
+ * @brief IEEE802.15.4 transceiver states
+ */
+typedef enum {
+    IEEE802154_TRX_STATE_TRX_OFF, /**< the transceiver state if off */
+    IEEE802154_TRX_STATE_RX_ON, /**< the transceiver is ready to receive packets */
+    IEEE802154_TRX_STATE_TX_ON, /**< the transceiver is ready to transmit a frame */
+} ieee802154_trx_state_t;
+
+/**
+ * @brief IEEE802.15.4 Radio HAL events
+ *
+ * To follow the IEEE802.15.4 convention, an event that responds to a Request
+ * is a confirmation (Confirm). Otherwise an Indication.
+ */
+typedef enum {
+    /**
+     * @brief the transceiver detected a valid SFD
+     */
+    IEEE802154_RADIO_INDICATION_RX_START,
+
+    /**
+     * @brief the transceiver received a frame and there's a frame in the
+     *        internal framebuffer.
+     *
+     * The transceiver is in @ref IEEE802154_TRX_STATE_RX_ON state when
+     * this funcion is called, but with framebuffer protection (either
+     * dynamic framebuffer protection or disabled RX). Thus, the frame
+     * won't be overwritten before calling the @ref ieee802154_radio_read
+     * function. However, @ref ieee802154_radio_read MUST be called in
+     * order to receive new packets. If there's no interest in the
+     * frame, the function can be called with a NULL buffer to drop
+     * the frame.
+     */
+    IEEE802154_RADIO_INDICATION_RX_DONE,
+
+    /**
+     * @brief the transceiver finished sending a frame or the
+     *        retransmission procedure
+     *
+     * If the radio supports frame retransmissions the 
+     * @ref ieee802154_radio_get_tx_status MAY be called to retrieve useful
+     * information (number of retries, frame pending bit, etc). The
+     * transceiver is in @ref IEEE802154_TRX_STATE_TX_ON state when this function
+     * is called.
+     */
+    IEEE802154_RADIO_CONFIRM_TX_DONE,
+    /**
+     * @brief the transceiver reports that the ACK timeout expired
+     *
+     * This event is present only if the radio support ACK timeout.
+     */
+    IEEE802154_RADIO_CONFIRM_ACK_TIMEOUT,
+    /**
+     * @brief the CCA procedure finished
+     */
+    IEEE802154_RADIO_CONFIRM_CCA,
+} ieee802154_trx_ev_t;
+
+/**
+ * @brief CSMA-CA exponential backoff parameters.
+ */
+typedef struct {
+    uint8_t min; /**< minimum value of the exponential backoff */
```

Comment:
Do we have the "default" values document and should reference them here?

Reply:
I think we define them in RIOT (min=3, retr=4, max=5). But not sure if the standard defines defaults

Reply:
I think this is out of the scope of the RDM though, since it's a MAC layer task

Reply:
I will mark it as resolved